sopel/plugins/rules.py

```diff
@@ -1158,36 +1166,63 @@ def is_global_rate_limited(self):
         rate_limit = datetime.timedelta(seconds=self._global_rate_limit)
         return self._metrics_global.is_limited(now - rate_limit)
 
+    def rate_limit_info(self, trigger: Trigger) -> RateLimitCheckInfo:
+        if trigger.admin or self.is_unblockable():
+            return RateLimitCheckInfo()
+
+        # TODO: these functions call now() and probably shouldn't
+        if self.is_user_rate_limited(trigger.nick):
+            template = self.get_user_rate_message(trigger.nick)
+        elif self.is_channel_rate_limited(trigger.sender):
+            template = self.get_channel_rate_message(trigger.nick, trigger.sender)
+        elif self.is_global_rate_limited():
+            template = self.get_global_rate_message(trigger.nick)
+        else:
+            return RateLimitCheckInfo()
+
+        # TODO: add:
+        #     * time_left
+        #     * time_left_sec
+        #     * rate_limit
+        #     * rate_limit_type
+        #     * command
+        #     * plugin
+        #     * ???
+        if template:
+            message = template.format(
+                nick=trigger.nick,
+                # TODO: add is_channel() to trigger?
+                channel=trigger.sender if not trigger.sender.is_nick() else None,
+                sender=trigger.sender,
+            )
+        else:
+            message = None
+
+        return RateLimitCheckInfo(message=message, limited=True)
+
     def get_user_rate_message(self, nick):
         template = self._user_rate_message or self._default_rate_message
 
         if not template:
             return None
 
-        return template.format(
-            nick=nick,
-        )
+        return template
 
     def get_channel_rate_message(self, nick, channel):
         template = self._channel_rate_message or self._default_rate_message
 
         if not template:
             return None
 
-        return template.format(
-            nick=nick,
-            channel=channel,
-        )
+        return template
 
     def get_global_rate_message(self, nick):
         template = self._global_rate_message or self._default_rate_message
 
         if not template:
             return None
 
-        return template.format(
-            nick=nick,
-        )
+        return template
```

Comment:
I was surprised there was no flake8 warning about unused method parameters on any of this. Then I remembered that #1765 exists, with a task item to add `flake8-unused-arguments` to the dev requirements someday.

Surely you just forgot to remove the now-unused positional arguments?

Reply:
Yea that's just oversight on my part as things churned around in this draft. These getters are removed in the latest revision and the `format()` lives on the `Sopel` side now.